cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -252,75 +244,68 @@ def validate_profile(profile: AzureProfile) -> bool:
     is_valid = True
     cli_tell(f"Profile name: {profile.name}")
 
-    # check invariants
     try:
-        raise_on_missing_fields(profile)
-        raise_on_invalid_storage(profile)
-    except (ProfilesInvalidError, ProfilesIncompleteError) as err:
-        cli_tell(str(err))
-        is_valid = False
+        # check invariants
+        validate_profile_configuration(profile)
 
-    # check if provided credentials allow to login to Azure
-    if has_complete_authentication_data(profile) and azure_login(profile):
-        # check if location is valid
-        if profile.location:
-            available_locations = list_locations()
-            if available_locations and profile.location not in available_locations:
-                cli_tell(f"The location is invalid in Azure: '{profile.location}'")
-                is_valid = False
+        # check if provided credentials allow to login to Azure
+        cred = profile_to_credentials(profile)
+        client = AzureClient.login_application(cred)
 
+        # check if location is valid
+        available_locations = client.list_locations()
+        if available_locations and profile.location not in available_locations:
```

Comment:
The test on `available_locations` being "non-null" is not necessary since it cannot be `None` and `not in` is true for any empty list (regardless of the object tested).

We though may want to report error if `AzureClient` returns no locations. We may also separately report the case when `profile.location` is "null" because it should not happen, right?

Reply:
updated
profile.location != None is checked along with other fields at the very beginning